releases.go

```diff
@@ -94,42 +102,82 @@ func (q ReleasesQuery) DefaultRange() headerutil.Range {
 	}
 }
 
-// ReleasesFind returns the first matching release.
-func (s *store) ReleasesFind(scope Scope) (*Release, error) {
+// releasesService is a service for creating and rolling back a Release.
+type releasesService struct {
+	*Empire
+}
+
+// ReleasesCreate creates a new release then submits it to the scheduler.
+func (s *releasesService) ReleasesCreate(ctx context.Context, db *gorm.DB, r *Release) (*Release, error) {
+	// Lock all releases for the given application to ensure that the
+	// release version is updated automically.
+	if err := db.Exec(`select 1 from releases where app_id = ? for update`, r.App.ID).Error; err != nil {
```

Comment:
Also found an issue where the previous lock we were obtaining wasn't sufficient to protect the atomicity of the Version of a release. See [TestEmpire_Deploy_Concurrent](https://github.com/remind101/empire/blob/395d91fc570b0e11a7553a899bb8e8c3a12b76f0/tests/empire/empire_test.go#L148-L206).